log_collector.h

```diff
@@ -0,0 +1,91 @@
+/*******************************************************************************
+The MIT License (MIT)
+
+Copyright (c) 2015 Dmitry "Dima" Korolev <dmitry.korolev@gmail.com>
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*******************************************************************************/
+
+#ifndef LOG_COLLECTOR_H
+#define LOG_COLLECTOR_H
+
+#include <string>
+#include <map>
+#include <ostream>
+
+#include "../Bricks/net/api/api.h"
+#include "../Bricks/time/chrono.h"
+#include "../Bricks/cerealize/cerealize.h"
+
+// An entry to log. Keeps everything that is easy to dig out. :-)
+struct LogEntry {
+  uint64_t t;                            // Unix epoch time in milliseconds.
+  std::string m;                         // HTTP method.
+  std::string u;                         // URL without fragments and query parameters.
+  std::map<std::string, std::string> q;  // URL query parameters.
+  std::string b;                         // HTTP body.
+  std::string f;                         // URL fragment.
+
+  // TODO(dkorolev): Add HTTP headers support to Bricks `net/api/types.h` HTTP client.
+  // std::map<std::string, std::string> h;  // Extra HTTP headers.
+
+  // TODO(dkorolev): Inbound IP address.
+  // TODO(dkorolev): Everything else we can/should think of.
+  // TODO(dkorolev): Resolve geolocation from IP?
```

Comment:
Yes, there are a few tools available out there. I'd lean towards the one we could compile into out binary, along with its data files, so that it just works.